### PR TITLE
New task to recursively store jet splittings

### PR DESCRIPTION
**Pull request by haandrew**
New task to run over data jets to store iteratively the jet splitting information

Applied suggested changers from maintainers

### CODE REVIEW

**Comment by mfasDa** [comment]
There was no need to close #3317, new commits are automatically added to existing requests.

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecursiveSoftDrop.cxx [hunk 1]
```diff
@@ -126,10 +126,13 @@ AliAnalysisTaskRecursiveSoftDrop::~AliAnalysisTaskRecursiveSoftDrop()
   //create a tree used for the MC data and making a 4D response matrix
   const char* nameoutput = GetOutputSlot(2)->GetContainer()->GetName();
   fTreeRecursive_Det = new TTree(nameoutput, nameoutput);
-  TString *fShapesVarNames_Det = new TString [nBranches];
   const char* nameoutput2 = GetOutputSlot(3)->GetContainer()->GetName();
   fTreeRecursive_True = new TTree(nameoutput2, nameoutput2);
-  TString *fShapesVarNames_True = new TString [nBranches];
+
+  const Int_t intBranches = 4;
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecursiveSoftDrop.cxx]: The real ALICE naming convention is constants starting with "k"

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecursiveSoftDrop.cxx [hunk 2]
```diff
@@ -126,10 +126,13 @@ AliAnalysisTaskRecursiveSoftDrop::~AliAnalysisTaskRecursiveSoftDrop()
...
+
+  std::vector<String> fShapeVarNames_Det(intBranches), nShapeVarNames_True(intBranches);
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecursiveSoftDrop.cxx]: The include for vector was not added.

**Comment by mfasDa** [comment]
+test

**Comment by haandrew** [comment]
Apologies. I was under the impression commits had to be squashed. Is this only when multiple commits are made prior to opening a pull request? I will update to adhere to the naming convention too and add the required include that will likely cause the +test to break

**Comment by mfasDa** [comment]
Yes, better squash commits in such case (however they were not squash). In case of interactive rebases you have to do a "git push -f ...", and also in this case the pull request is automatically updated for new commits in the same branch.

**Comment by mfasDa** [comment]
You forgot to add the task to the linkdef file.

**Comment by haandrew** [comment]
Hi Markus, I have made the changes and done a local build test and it should work fine now. Are you sure I am missing the task from the linkdef file, are you referring to PWGJEEMCALJetTasksLinkDef.h because that has been modified in the commit

**Comment by raymondEhlers** [comment]
+test

**Comment by raymondEhlers** [comment]
After a brief glance, I'd say it looks okay now, but I'll leave it to someone else to confirm and approve

**Comment by mfasDa** [comment]
+1

**Pull request merged**